test/unit/math/prim/fun/quad_form_sym_test.cpp

```diff
@@ -0,0 +1,74 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathMatrixPrim, quad_form_sym_mat) {
```

Comment:
This should be testing boundary conditions of size 1 and size 0 matrices as well as these bigger cases as the other quad_form tests do.